pkg/buildcontext/file.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildcontext
+
+import (
+	"os"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/constants"
+	"github.com/GoogleContainerTools/kaniko/pkg/util"
+)
+
+// File unifies calls to download and unpack the build context.
+type File struct {
+	context string
+}
+
+// UnpackTarFromBuildContext untar the tar file
+func (f *File) UnpackTarFromBuildContext() (string, error) {
+	directory := constants.BuildContextDir
+	if err := os.MkdirAll(directory, 0750); err != nil {
+		return "", err
```

Comment:
please wrap this error message 
` return "", errors.Warp(err, "unpacking tar from build context") `